### PR TITLE
Joseph's dotnet6 - With some adjustments

### PR DESCRIPTION
**Pull request by jasonleenaylor**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)

### CODE REVIEW

**Review by ermshiperete — CHANGES_REQUESTED** [review]

File: src/SIL.LCModel.Core/GenerateKernelCs.proj
Comment (jasonleenaylor) [reply src/SIL.LCModel.Core/GenerateKernelCs.proj:27]: @josephmyers @papeh 
Oh yes, I forgot about this little hack. And we didn't discuss it in our meeting this morning. We need a different solution to this before we can merge this PR. 😄

Comment (ermshiperete) [reply src/SIL.LCModel.Core/GenerateKernelCs.proj:27]: @jasonleenaylor Where do we need it? Can we update the component that needs it?

**Comment by jasonleenaylor** [comment]
*[`src/SIL.LCModel.Build.Tasks/SIL.LCModel.Build.Tasks.csproj` line 17 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYi-r8-17:-N6TFPFk0VBLJ0fYFz8K:b-ezqz2u) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Build.Tasks/SIL.LCModel.Build.Tasks.csproj#L17)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

I just created a release for SIL.IdlImporter, so we should depend on that. Do we need to have the dependency on SIL.IdlImporter in the nuget package, or should `PrivateAssets=All` be re-added?
</blockquote></details>

We can probably re-add the PrivateAssets=All, this was an experiment to try and solve the issue resolving Microsoft.Build.Utilities before I gave up all hope and added the hack of directly copying it. Now that we know this build works on all windows developer scenarios I hope to find a more principled solution to that problem.

**Comment by jasonleenaylor** [comment]
*[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 27 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYj-r8-27:-N6TIeJE77JYMUvbtNyX:b-i4oxn2) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Core/GenerateKernelCs.proj#L27)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

@jasonleenaylor Where do we need it? Can we update the component that needs it?
</blockquote></details>

The IdlImp build task needs it. My best guess as to why this fails is that when we start multi-targeting we lose assembly loading from the GAC during the build.

**Comment by jasonleenaylor** [comment]
*[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 27 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYj-r8-27:-N6TJUH54Jp6gRqf7jyn:bn1j43) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Core/GenerateKernelCs.proj#L27)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

The IdlImp build task needs it. My best guess as to why this fails is that when we start multi-targeting we lose assembly loading from the GAC during the build.
</blockquote></details>

Without that copy you get this, and nothing else I tried could get me around it.

```
error MSB4062: The "IdlImp" task could not be loaded from the assembly C:\Repositories\liblcm\/artifacts/Debug/net461\SIL.LCModel.Build.Tasks.dll. Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0
```

I tried adding a specific reference to get it to copy the file and it found a reference only assembly that failed at runtime. Any and all ideas are welcome.

**Review by papeh — COMMENTED** [review]
> Reviewed 2 of 18 files at r1, 8 of 22 files at r7.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: 10 of 31 files reviewed, 2 unresolved discussions (waiting on @ermshiperete, @josephmyers, and @papeh)

___
*[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 27 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYj-r8-27:-N6TTO9bBd7bKGactFP-:b4s4oh3) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Core/GenerateKernelCs.proj#L27)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

Without that copy you get this, and nothing else I tried could get me around it.

```
error MSB4062: The "IdlImp" task could not be loaded from the assembly C:\Repositories\liblcm\/artifacts/Debug/net461\SIL.LCModel.Build.Tasks.dll. Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0
```

I tried adding a specific reference to get it to copy the file and it found a reference only assembly that failed at runtime. Any and all ideas are welcome.
</blockquote></details>

Check the .dll in to our repository?
`Condition="'$(OS)'=='Windows_NT'"`?

**Review by papeh — COMMENTED** [review]
> Reviewed 5 of 18 files at r1, 1 of 1 files at r2, 7 of 22 files at r7, 1 of 1 files at r8, 2 of 2 files at r9.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: 26 of 31 files reviewed, 2 unresolved discussions (waiting on @ermshiperete, @josephmyers, and @papeh)

___
*[`src/SIL.LCModel.Core/SIL.LCModel.Core.csproj` line 137 at r9](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6Ti-mg4StjMoKl7286:-N6Ti-mg4StjMoKl7287:bpbrjl3) ([raw file](https://github.com/sillsdev/liblcm/blob/f55f8dfaf3f06745de2fc8bee4809b0fe9cd2b1c/src/SIL.LCModel.Core/SIL.LCModel.Core.csproj#L137)):*
> ```XML
> 
>   <Target Name="GenerateIcuData" AfterTargets="CoreBuild" Inputs="@(IcuDataInputs)" Outputs="@(IcuDataOutputs)" >
>     
> ```

This count could be omitted. Having the text and condition in the same order for all errors would improve readability.



_Code quote (i):_
```XML
two
```



_Code snippet (ii):_
```XML
three
```

**Comment by jasonleenaylor** [comment]
*[`src/SIL.LCModel.Utils/SIL.LCModel.Utils.csproj` line 15 at r9](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6TjkJU-Q8_j3MFoL2U:-N6Tv6NYCfbVu_BwNRFf:b7nmj12) ([raw file](https://github.com/sillsdev/liblcm/blob/f55f8dfaf3f06745de2fc8bee4809b0fe9cd2b1c/src/SIL.LCModel.Utils/SIL.LCModel.Utils.csproj#L15)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

¿¿¿should FW also be updated to Netwonsoft.Json 13??? 
</blockquote></details>

Hmm. 
Probably.

**Comment by ermshiperete** [comment]
> _[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 27 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYj-r8-27:-N6TJUH54Jp6gRqf7jyn:bn1j43) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Core/GenerateKernelCs.proj#L27)):_
> Previously, jasonleenaylor (Jason Naylor) wrote…
> 
> Without that copy you get this, and nothing else I tried could get me around it.
> 
> ```
> error MSB4062: The "IdlImp" task could not be loaded from the assembly C:\Repositories\liblcm\/artifacts/Debug/net461\SIL.LCModel.Build.Tasks.dll. Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0
> ```
> 
> I tried adding a specific reference to get it to copy the file and it found a reference only assembly that failed at runtime. Any and all ideas are welcome.

Do we have to build SIL.LCModel.Build.Tasks for net461 or would netstandard2.0 be sufficient? (If we remove net461 we'll also have to change SIL.LCModel.Build.Tasks.props - that mentions net461 as well)

**Comment by jasonleenaylor** [comment]
*[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 27 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYj-r8-27:-N6U6ppl2vyXie9KVJH_:b-an2brp) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Core/GenerateKernelCs.proj#L27)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

Check the .dll in to our repository?
`Condition="'$(OS)'=='Windows_NT'"`?
</blockquote></details>

Thanks to Eberhard's suggestion about using the netstandard2.0 version we have a new hack to avoid the copy that should also work on Linux.

**Comment by jasonleenaylor** [comment]
Yes we do, I'm going to have to play around with this in a Flex build today but your question led me to a much improved hack.

**Comment by jasonleenaylor** [comment]
*[`src/SIL.LCModel.Build.Tasks/SIL.LCModel.Build.Tasks.csproj` line 17 at r8](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6PXaKhpNeSxjC9-mYi-r8-17:-N6U7xh7EB7t5iBb4JVN:b-896fix) ([raw file](https://github.com/sillsdev/liblcm/blob/75c5af5a4d95e8f61714f1fcc58404604bdb2776/src/SIL.LCModel.Build.Tasks/SIL.LCModel.Build.Tasks.csproj#L17)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

We can probably re-add the PrivateAssets=All, this was an experiment to try and solve the issue resolving Microsoft.Build.Utilities before I gave up all hope and added the hack of directly copying it. Now that we know this build works on all windows developer scenarios I hope to find a more principled solution to that problem.
</blockquote></details>

Done.

**Review by papeh — COMMENTED** [review]
:lgtm: when angle bracket is rounded

> Reviewed 1 of 8 files at r6, 3 of 22 files at r7, 1 of 1 files at r10, 2 of 2 files at r11.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238#-:-N6UETs53tJPeS8sPdkk:biyei7n)* status: all files reviewed (commit messages unreviewed), 3 unresolved discussions (waiting on @ermshiperete, @jasonleenaylor, and @josephmyers)

___
*[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 26 at r11](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6UE1N_6X4AnPGbjpXN:-N6UE1N_6X4AnPGbjpXO:undefined) ([raw file](https://github.com/sillsdev/liblcm/blob/4f3d1498a8103ac699fdb4910b64e67c17a3e286/src/SIL.LCModel.Core/GenerateKernelCs.proj#L26)):*
> ```XML
> 		<UsingNamespaces Include="SIL.LCModel.Utils" />
> 	</ItemGroup>
> 	<Target Name="GenerateKernelCs" Inputs="@(KernelIdhFiles>" Outputs="$(OutDir)KernelInterfaces/FwKernelTlb.json">
> ```





_Code quote (i):_
```XML
@(KernelIdhFiles>
```



_Code snippet (ii):_
```XML
@(KernelIdhFiles)
```

**Review by jasonleenaylor — COMMENTED** [review]
> Dismissed @ermshiperete from a discussion.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: 30 of 31 files reviewed, 2 unresolved discussions (waiting on @josephmyers and @papeh)

___
*[`src/SIL.LCModel.Core/GenerateKernelCs.proj` line 26 at r11](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6UE1N_6X4AnPGbjpXN:-N6UKb6h1i5nFL0QLI0S:b-896fix) ([raw file](https://github.com/sillsdev/liblcm/blob/4f3d1498a8103ac699fdb4910b64e67c17a3e286/src/SIL.LCModel.Core/GenerateKernelCs.proj#L26)):*
> ```XML
> 		<UsingNamespaces Include="SIL.LCModel.Utils" />
> 	</ItemGroup>
> 	<Target Name="GenerateKernelCs" Inputs="@(KernelIdhFiles>" Outputs="$(OutDir)KernelInterfaces/FwKernelTlb.json">
> ```

Done.

**Review by ermshiperete — APPROVED** [review]

**Review by papeh — APPROVED** [review]
> Reviewed 1 of 1 files at r12.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: all files reviewed (commit messages unreviewed), 1 unresolved discussion (waiting on @josephmyers)

**Review by josephmyers — APPROVED** [review]
> Reviewed 6 of 18 files at r1, 1 of 1 files at r2, 18 of 22 files at r7, 1 of 1 files at r8, 1 of 2 files at r9, 2 of 2 files at r11, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: all files reviewed, 1 unresolved discussion (waiting on @ermshiperete)

**Review by papeh — APPROVED** [review]
> Reviewed 5 of 7 files at r13, 2 of 2 files at r14, 1 of 1 files at r15, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: all files reviewed, 2 unresolved discussions (waiting on @ermshiperete and @jasonleenaylor)

___
*[`.github/workflows/ci-cd.yml` line 38 at r14](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6nosIP4Mdr9S9dSm2P:-N6nosIP4Mdr9S9dSm2Q:b-pntkyh) ([raw file](https://github.com/sillsdev/liblcm/blob/bae1b9e9b26ef632d2979c972ef0d805b27e5aa1/.github/workflows/ci-cd.yml#L38)):*
> ```YAML
>     - name: Install NUnit.ConsoleRunner
>       run: nuget install NUnit.ConsoleRunner -Version 3.13.0 -DirectDownload -OutputDirectory .
>         
> ```

nit: trailing whitespace
___
*[`tests/SIL.LCModel.Utils.Tests/SIL.LCModel.Utils.Tests.csproj` line 21 at r15](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6npiq_9hPUjoLJOlBF:-N6npiq_9hPUjoLJOlBG:b-43kivc) ([raw file](https://github.com/sillsdev/liblcm/blob/bb90d7c8cee154d836c759a766bd00a280ca49f0/tests/SIL.LCModel.Utils.Tests/SIL.LCModel.Utils.Tests.csproj#L21)):*
> ```XML
>   </ItemGroup>
>   <ItemGroup Condition="'$(TargetFramework)' == 'net461'">
>     <PackageReference Include="Mono.Posix" Version="5.4.0.201" PrivateAssets="All" />
> ```

DRY code makes me happy :smile:

File: .github/workflows/ci-cd.yml
Comment (ermshiperete) [inline .github/workflows/ci-cd.yml:16]: So we don't want to build on Linux?

File: .github/workflows/ci-cd.yml
Comment (ermshiperete) [inline .github/workflows/ci-cd.yml:44]: Would `dotnet test` work here?

**Review by ermshiperete — APPROVED** [review]

**Review by papeh — APPROVED** [review]
> Reviewed 1 of 1 files at r16, 1 of 1 files at r18, 1 of 1 files at r19, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: all files reviewed, 3 unresolved discussions (waiting on @ermshiperete and @jasonleenaylor)

File: .github/workflows/ci-cd.yml
Comment (ermshiperete) [inline .github/workflows/ci-cd.yml:54]: Shouldn't this be `artifacts/*.nupkg` ?

```suggestion
      run: dotnet nuget push artifacts/*.nupkg -s https://api.nuget.org/v3/index.json -k ${{secrets.SILLSDEV_PUBLISH_NUGET_ORG}} --skip-duplicate
```

File: .github/workflows/ci-cd.yml
Comment (ermshiperete) [inline .github/workflows/ci-cd.yml:38]: We'll have to add llso for `icu-fw` to be available. That can be done with:

```
wget -qO - http://linux.lsdev.sil.org/downloads/sil-testing.gpg | sudo tee /etc/apt/trusted.gpg.d/linux-lsdev-sil-org.asc
sudo add-apt-repository "deb http://linux.lsdev.sil.org/ubuntu $(lsb_release -sc)-experimental main"
```

**Comment by ermshiperete** [comment]
At least some of the test failures might be caused by https://github.com/sillsdev/icu-dotnet/issues/136

**Review by papeh — COMMENTED** [review]
> Reviewed 24 of 28 files at r21, 2 of 3 files at r22, 1 of 2 files at r24, 1 of 5 files at r31, 4 of 4 files at r32, all commit messages.
> Dismissed @ermshiperete from 3 discussions.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: all files reviewed, 3 unresolved discussions (waiting on @ermshiperete and @jasonleenaylor)

___
*[`.github/workflows/ci-cd.yml` line 16 at r19](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6nVE19qPq3M38K2ggu-r19-16:-N7Gw3Zc-THXJldCC_zl:b-d4dpwj) ([raw file](https://github.com/sillsdev/liblcm/blob/4bd40bbde13887a66a7110524edafc8e030196f6/.github/workflows/ci-cd.yml#L16)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

So we don't want to build on Linux?
</blockquote></details>

The current revision builds on both platforms, as desired.
___
*[`.github/workflows/ci-cd.yml` line 44 at r19](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6nVE19qPq3M38K2ggu-r19-44:-N7GwRs-B2_5YaIjO_iq:bb6un1j) ([raw file](https://github.com/sillsdev/liblcm/blob/4bd40bbde13887a66a7110524edafc8e030196f6/.github/workflows/ci-cd.yml#L44)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

Would `dotnet test` work here?
</blockquote></details>

Done
___
*[`.github/workflows/ci-cd.yml` line 38 at r25](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6nVE19qPq3M38K2ggu-r25-38:-N71hydZ7-pctCedaLKe:boyexbg) ([raw file](https://github.com/sillsdev/liblcm/blob/e32bf6cae05dd22b10db043b60a8e57331eb3a5f/.github/workflows/ci-cd.yml#L38)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

We'll have to add llso for `icu-fw` to be available. That can be done with:

```
wget -qO - http://linux.lsdev.sil.org/downloads/sil-testing.gpg | sudo tee /etc/apt/trusted.gpg.d/linux-lsdev-sil-org.asc
sudo add-apt-repository "deb http://linux.lsdev.sil.org/ubuntu $(lsb_release -sc)-experimental main"
```
</blockquote></details>

Done. Thanks.
___
*[`.github/workflows/ci-cd.yml` line 54 at r25](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6nVE19qPq3M38K2ggu-r25-54:-N71i-hZ08tT35S8KN6Y:bb6un1j) ([raw file](https://github.com/sillsdev/liblcm/blob/e32bf6cae05dd22b10db043b60a8e57331eb3a5f/.github/workflows/ci-cd.yml#L54)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

Shouldn't this be `artifacts/*.nupkg` ?

```suggestion
      run: dotnet nuget push artifacts/*.nupkg -s https://api.nuget.org/v3/index.json -k ${{secrets.SILLSDEV_PUBLISH_NUGET_ORG}} --skip-duplicate
```
</blockquote></details>

Done
___
*[`.github/workflows/ci-cd.yml` line 66 at r31](https://reviewable.io/reviews/sillsdev/liblcm/238#-N7GwjxH1oYpbFBmAcpR:-N7GwjxH1oYpbFBmAcpS:b-5juz2t) ([raw file](https://github.com/sillsdev/liblcm/blob/15e848d566436519be791b462afe6d18fb5d366b/.github/workflows/ci-cd.yml#L66)):*
> ```YAML
>         name: NugetPackages
>         path: artifacts/*.nupkg
>       if: github.event_name == 'pull_request'
> ```

Does this publish artifacts from pull requests? It could use a comment explaining what it does and why it is needed.

File: .github/workflows/ci-cd.yml
Comment (ermshiperete) [inline .github/workflows/ci-cd.yml:59]: Do we have to check the branch name as well so that we don't publish to nuget on a push to a feature branch?

**Review by ermshiperete — CHANGES_REQUESTED** [review]
We should remove `appveyor.yml` file

**Review by papeh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: 36 of 38 files reviewed, 5 unresolved discussions (waiting on @ermshiperete, @jasonleenaylor, @josephmyers, and @papeh)

___
*[`.github/workflows/ci-cd.yml` line 59 at r32](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6nVE19qPq3M38K2ggu-r32-59:-N7H8-jxB3oGl6wAj42i:b5mrhj3) ([raw file](https://github.com/sillsdev/liblcm/blob/1532c43b7223d7bc9417fb5c76631e67a6f3134f/.github/workflows/ci-cd.yml#L59)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

Do we have to check the branch name as well so that we don't publish to nuget on a push to a feature branch?
</blockquote></details>

That's checked on lines 5-8

**Review by papeh — COMMENTED** [review]
> Dismissed @ermshiperete from a discussion.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: 36 of 38 files reviewed, 4 unresolved discussions (waiting on @ermshiperete, @jasonleenaylor, @josephmyers, and @papeh)

___
*[`src/SIL.LCModel.Utils/MiscUtils.cs` line 845 at r32](https://reviewable.io/reviews/sillsdev/liblcm/238#-N6sg0Ul57PniaTbRoW1-r32-845:-N7H9oF48RuKfR-JF7kH:bb7gzl0) ([raw file](https://github.com/sillsdev/liblcm/blob/1532c43b7223d7bc9417fb5c76631e67a6f3134f/src/SIL.LCModel.Utils/MiscUtils.cs#L845)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

Is this line intentional or a remnant of getting things to build? If it is intentional we should probably change it to `Debug.WriteLine`.
</blockquote></details>

gone

**Review by ermshiperete — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]
> Reviewed 2 of 2 files at r33, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: all files reviewed, 1 unresolved discussion (waiting on @jasonleenaylor)

**Review by papeh — APPROVED** [review]
> Reviewed 2 of 2 files at r33, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/238)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jasonleenaylor)

**Pull request merged**
